edge-e2e/wrapper/nodejs-server-server/package.json

```diff
@@ -0,0 +1,29 @@
+{
+  "name": "iot-sdk-device-client-rest-api",
+  "version": "1.0.0",
+  "description": "REST API definition for End-to-end testing of the Azure IoT SDKs.  All SDK APIs that are tested by our E2E tests need to be defined in this file.  This file takes some liberties with the API definitions.  In particular, response schemas are undefined, and error responses are also undefined.",
+  "main": "index.js",
+  "scripts": {
+    "prestart": "npm install",
+    "start": "node index.js"
+  },
+  "keywords": [
+    "swagger"
+  ],
+  "license": "Unlicense",
+  "private": true,
+  "dependencies": {
+    "async": "^2.6.1",
+    "azure-event-hubs": "0.0.8",
+    "azure-iot-common": "1.7.1",
+    "azure-iot-device": "1.7.1",
+    "azure-iot-device-amqp": "1.7.1",
+    "azure-iot-device-http": "1.7.1",
+    "azure-iot-device-mqtt": "1.7.1",
+    "azure-iothub": "1.7.1",
+    "bluebird": "^3.3.0",
```

Comment:
>"bluebird": "^3.3.0", [](start = 4, length = 21)

since we control the environment we should be able to assume it'll have a global Promise object and not need external libraries

Reply:
You'll have to explain this a little better.  It's also worth mentioning that some of the code that relies on Bluebird is auto-generated, but I'm not sure how that affects your comment.

---
In reply to: [210696646](https://github.com/Azure/azure-iot-sdk-node/pull/346#discussion_r210696646) [](ancestors = 210696646)

Reply:
if the autogenerated code relies on bluebird then bluebird is probabyl a dependency of swagger-tools or one of the other packages. the point is that we don't use bluebird in our SDK anymore: we rely on the fact that node comes with a global `Promise` object and that it's the one we should be using.

---
In reply to: [212857482](https://github.com/Azure/azure-iot-sdk-node/pull/346#discussion_r212857482) [](ancestors = 212857482,210696646)